geomesa-fs/geomesa-fs-storage/geomesa-fs-storage-common/src/main/scala/org/locationtech/geomesa/fs/storage/common/FileMetadata.scala

```diff
@@ -139,6 +145,40 @@ class FileMetadata private (fc: FileContext,
       FileMetadata.save(this)
     }
   }
+
+  var internalCount = data.map(_.getInt("count")).getOrElse(0)
+  var bounds: Envelope = data.map(_.getDoubleList("bounds")).map { doubles =>
+    new ReferencedEnvelope(doubles(0), doubles(1), doubles(2), doubles(3), CRS_EPSG_4326)
+  }.orNull
+
+  override def getFeatureCount: Int = internalCount
+  override def increaseFeatureCount(count: Int): Unit = {
+    internalCount += count
+  }
+
+  override def getEnvelope: ReferencedEnvelope =
+    if (bounds == null) {
+      ReferencedEnvelope.EVERYTHING
```

Comment:
i think envelopes have an idea of a 'null' or empty envelope - I think we could avoid having bounds be a var by just setting it to the empty envelope to start with. would avoid the dreaded 'double mutability' (e.g. `var m = mutable.Map.empty`)